manager/dispatcher/dispatcher.go

```diff
@@ -584,6 +584,24 @@ func (d *Dispatcher) UpdateTaskStatus(ctx context.Context, r *api.UpdateTaskStat
 		validTaskUpdates = append(validTaskUpdates, u)
 	}
 
+	// NOTE(dperny): So, there is an issue here, a bad design decision. We add
```

Comment:
nit: This is a great summary, but maybe adjust it  so it feels more like "how it works" rather than "what you did".

Reply:
I've reworded this slightly from when you made this comment, but not changed the specific line, which is why this looks unaddressed.